swarm/network/hive.go

```diff
@@ -44,7 +44,9 @@ type Overlay interface {
 	On(OverlayConn) (depth uint8, changed bool)
 	Off(OverlayConn)
 	// register peer addresses
-	Register([]OverlayAddr) error // used by the
+	Register([]OverlayAddr) error
+	// unregister peer address
+	Unregister(OverlayPeer)
```

Comment:
this should be `OverlayAddr`

Reply:
OK, then I need to figure out how to get `OverlayAddr` from `p` (which is `*p2p.Peer`). We tried to do that, but were unsuccessful for some reason.